tests/aws/services/stepfunctions/templates/scenarios/statemachines/map_state_config_distributed_reentrant.json5

```diff
@@ -0,0 +1,61 @@
+{
+  "Comment": "MAP_STATE_CONFIG_DISTRIBUTED_REENTRANT",
+  "StartAt": "StartState",
+  "States": {
+    // Populate memory with two fields: number of iterations left, and input values.
+    "StartState": {
+      "Type": "Pass",
+      "Parameters": {
+        "Iterations": 3,
+        "Values.$": "States.ArrayRange(0, 3, 1)"
+      },
+      "Next": "BeforeIteration"
+    },
+    // Prepare the iteration by updating the iterations count.
+    "BeforeIteration": {
+      "Type": "Pass",
+      "Parameters": {
+        "Iterations.$": "States.MathAdd($.Iterations, -1)",
+        "Values.$": "$.Values"
+      },
+      "Next": "IterationBody"
+    },
+    // Run a distributed map state on the values field.
+    "IterationBody": {
+      "Type": "Map",
+      "MaxConcurrency": 1,
+      "ItemsPath": "$.Values",
+      "ItemProcessor": {
```

Comment:
This `ItemProcessor` is the only change compared to the legacy version. Maybe a comment could help to point out this difference because I didn't know the difference between legacy and this one without doing a diff.